validation-test/compiler_crashers_2_fixed/sr11027.swift

```diff
@@ -0,0 +1,20 @@
+// This source file is part of the Swift.org open source project
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+// RUN: not %target-swift-frontend %s -emit-silgen
+
+func sr11027(_ f: @convention(block) @autoclosure () -> Int) -> Void {}
+sr11027(1)
+
+func sr11027_c(_ f: @convention(c) @autoclosure () -> Int) -> Void {}
+sr11027_c(1)
+
+func sr11027_swift(_ f: @convention(swift) @autoclosure () -> Int) -> Void {} // OK
+sr11027_swift(1)
+
+func sr11027_thin(_ f: @convention(thin) @autoclosure () -> Int) -> Void {} // OK
```

Comment:
Could you also add test cases for `method`, `objc_method` and `witness_method`?

Reply:
Those types are invalid here, they are only valid at SIL level when it is performing validation on `SILType`. So not sure, if it would be useful add them here.

Reply:
Hmm if we already diagnose them when used in Swift code then it's fine.